### PR TITLE
Mongoid 7 compatibility

### PR DESCRIPTION
**Pull request by tomasc**

### CODE REVIEW

**Comment by tomasc** [comment]
@dblock seems like the tests fail for Mongoid 3 & 4. If you check the messages, the results differ by ids – not by values they actual test. Not sure why …

As a quick solution, would you be ok making the new version of this gem Mongoid >= 5 compatible? ([First version of 5 has been released back in 2015](https://rubygems.org/gems/mongoid/versions).)

**Comment by dblock** [comment]
mongoid-compatibility added `_or_newer?` method in 0.5.1, so that needs a `>=` in gemspec as well

**Comment by dblock** [comment]
Is the problem purging data between tests for older versions of mongoid? Maybe swap `Mongoid.purge!` for database_cleaner?

**Comment by tomasc** [comment]
@dblock switching to `database_cleaner` helped a bit, but still failing on older versions of Mongoid. Any idea why could that be?

**Comment by dblock** [comment]
They look like legit errors because of something fixed in a newer version of MongoDB that's being used, the index name needs to be unique. I will take a look if you can't figure it out.

**Comment by dblock** [comment]
I spent a fair amount of time trying to figure out what's going on with these specs and I failed to make any sense of it. Rebase this against https://github.com/mongoid/mongoid_fulltext/pull/39 which just marks the two failing specs `pending` for older versions of mongoid instead of removing support for those versions and lets call it a day. Note that I cannot reproduce the failures on my local OSX with all the same versions of MongoDB, Ruby, etc.

**Comment by tomasc** [comment]
@dblock thank you, i have rebased, now we're all green again

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -11,19 +12,27 @@
...
 Mongoid.configure do |config|
   config.connect_to('mongoid_fulltext_test')
 end
 
+Mongoid.logger.level = Logger::INFO
+Mongo::Logger.logger.level = Logger::INFO if Mongoid::Compatibility::Version.mongoid5_or_newer?
+
+DatabaseCleaner.orm = :mongoid
+DatabaseCleaner.strategy = :truncation
+
```
Comment (dblock) [inline spec/spec_helper.rb]: The 4 lines above are here twice, bad copy-paste ;)

**Comment by dblock** [comment]
Merged

**Pull request merged**